### PR TITLE
docs: update theming `styles` section

### PR DESCRIPTION
**Pull request by levithomason**
This PR updates the theming guide:

- Update `styles` info to match the flow and pacing of the `variables` section:
    - show component instance usage via the `styles` prop on the root
    - show component instance usage via the `styles` prop on a named slot
    - show multiple component usage via the `Provider`
    - give info on styling the entire app via the `Provider`
- Replace references to "component anatomy" with "component slots".
- Clean up verbiage and grammar.

# Before
image

# After
image

### CODE REVIEW

File: docs/src/views/Theming.tsx [hunk 1]
```diff
@@ -107,8 +106,8 @@ export default () => (
       )}
     />
     <p>
-      You can customize all components in your app by defining component variables on a Provider at
-      the root of your app.
+      You can customize component variables for your entire app by defining component variables on a{' '}
+      <NavLink to="components/provider">Provider</NavLink> at the root of your app.
```
Comment (kuzhelov) [inline docs/src/views/Theming.tsx:110]: 👍

File: docs/src/views/Theming.tsx [hunk 2]
```diff
@@ -121,42 +120,92 @@ export default () => (
...
+    <p>
+      Every slot (named part) of every component also accepts <code>styles</code> that are applied
+      to the root of the slot.
     </p>
-
     <ExampleSnippet
       value={[
         `<Button`,
-        `  icon={{ name: "user", styles: { borderBottom: '4px solid red' } }}`,
+        `  icon={{ name: 'user', styles: { boxShadow: '0 0 0 2px red' } }}`,
```
Comment (kuzhelov) [inline docs/src/views/Theming.tsx:142]: very representative both logically and visually 👍

File: docs/src/views/Theming.tsx [hunk 3]
```diff
@@ -121,42 +120,92 @@ export default () => (
...
-      You can style the <code>icon</code> part of a <code>Button</code> component.
+      You can define <code>styles</code> on a single component instance.
+    </p>
+    <ExampleSnippet
+      value={`<Text styles={{ color: 'green' }}>This is green text</Text>`}
+      render={() => <Text styles={{ color: 'green' }}>This is green text</Text>}
+    />
+    <p>
+      Every slot (named part) of every component also accepts <code>styles</code> that are applied
+      to the root of the slot.
```
Comment (kuzhelov) [inline docs/src/views/Theming.tsx]: `root` -> `root element`?

Comment (levithomason) [reply docs/src/views/Theming.tsx]: Indeed, updating...

File: docs/src/views/Theming.tsx [hunk 4]
```diff
@@ -121,42 +120,92 @@ export default () => (
...
+          >
+            <span>
+              <Button icon="user" content="Profile" />
+              <Button icon="user" content="Profile" />
+            </span>
+          </Provider>
+        </div>
+      )}
     />
+    <p>
```
Comment (kuzhelov) [inline docs/src/views/Theming.tsx:206]: just as a side notice - currently there are no links to the next (if necessary, as there are components sections coming next) and (surely) previous docs page - the ones that are present for the rest of the docs pages. Most probably you already have a plan to address that

**Review by kuzhelov — APPROVED** [review]
👍

**Pull request merged**